wagtail/admin/views/reports/page_types_usage.py

```diff
@@ -0,0 +1,175 @@
+import django_filters
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.exceptions import PermissionDenied, ValidationError
+from django.db.models import Count, F, OuterRef, Q, Subquery
+from django.utils.translation import gettext_lazy as _
+
+from wagtail.admin.filters import WagtailFilterSet
+from wagtail.admin.views.reports import ReportView
+from wagtail.coreutils import get_content_languages
+from wagtail.models import ContentType, Page, Site, get_page_models
+from wagtail.permission_policies.pages import PagePermissionPolicy
+from wagtail.users.utils import get_deleted_user_display_name
+
+
+def _get_locale_choices():
+    choices = [  # noqa: C416
+        (language_code, display_name)
+        for language_code, display_name in get_content_languages().items()
+    ]
+    return choices
+
+
+def _get_site_choices():
+    """Tuples of (site root page path, site display name) for all sites in project."""
+    choices = [(site.root_page.path, str(site)) for site in Site.objects.all()]
+    return choices
+
+
+def _annotate_last_edit_info(queryset, language_code, site_root_path):
+    latest_edited_page_filter_kwargs = {}
+    page_count_filter_kwargs = {}
+    if language_code:
+        latest_edited_page_filter_kwargs["locale__language_code"] = language_code
+        page_count_filter_kwargs["pages__locale__language_code"] = language_code
+    if site_root_path:
+        latest_edited_page_filter_kwargs["path__startswith"] = site_root_path
+        page_count_filter_kwargs["pages__path__startswith"] = site_root_path
+
+    latest_edited_page = Page.objects.filter(
+        content_type=OuterRef("pk"), **latest_edited_page_filter_kwargs
+    ).order_by(F("latest_revision_created_at").desc(nulls_last=True), "title", "-pk")
+
+    queryset = queryset.annotate(
+        count=Count("pages", filter=Q(**page_count_filter_kwargs)),
+        last_edited_page_id=Subquery(latest_edited_page.values("pk")[:1]),
+        last_edited_page_owner_id=Subquery(latest_edited_page.values("owner__pk")[:1]),
```

Comment:
In queries, you can use the column automatically added by a `ForeignKey` when you don't need anything else from the other table, and the query will be a little simpler as a result:
```suggestion
        last_edited_page_owner_id=Subquery(latest_edited_page.values("owner_id")[:1]),
```